### PR TITLE
Buffer incoming `SinkRecords` before sending batches

### LINKED ISSUE #133: Configurable buffering of incoming records
We need to offer users a way to configure how much data we will buffer in the connector before making Batch API submissions, and for how long we will wait for the buffer to fill (as a maximum) before making a submission. This is needed because:

* Kafka Connect appears to be calling `SinkTask.put()` very frequently in testing (every 10ms), with very low numbers of records per call.
* There's no way to configure this behaviour directly. Only the maximum poll interval can be controlled through the `max.poll.interval.ms` Kafka Connect property. This defaults to 5 minutes, but we can see in testing the actual poll interval will typically be much smaller.
* Incoming rates of records from Kafka and sizes of the outgoing Ably messages will vary by workload. Requirements on the latency of outgoing messages will also vary by workload. Users will need a way to configure the Connector in a way that achieves the best compromise of throughput, latency and cost (through API consumption) for their workload.

Kafka Connect *does* offer direct control over the interval at which `SinkTask.flush()` is called, but this doesn't really help us. The expected behaviour of `flush()` is that any outstanding operations are concluded so that offsets can be committed. If we buffer content from each `put()` call but don't submit any Batch API calls until the next `flush()`, we're spending a lot of time idle that could have been spent productively sending messages in parallel.

The [documentation](https://kafka.apache.org/documentation.html#connect_sinktasks) from implementing `SinkTask` suggests that internal buffering of records provided by `put()` is expected and common:

> The `put()` method should contain most of the implementation, accepting sets of SinkRecords, performing any required translation, and storing them in the destination system. This method does not need to ensure the data has been fully written to the destination system before returning. In fact, in many cases internal buffering will be useful so an entire batch of records can be sent at once, reducing the overhead of inserting events into the downstream data store.

It'll be far simpler for users if we expose direct controls over this buffering behaviour than documenting how to approximate parts of it using various Kafka properties, and the ongoing support burden will be much lower.

### PR DESCRIPTION
**Pull request by jaley**
This PR adds buffering of incoming records before sending batches to Ably. It can be configured to send batches when:

* a maximum buffer size is reached, OR
* a maximum delay has expired

... whichever happens first.

This is needed because we're seeing in initial load testing that `SinkTask.put()` calls happen with a very high frequency (about 10ms in testing) and the user has no direct control over this. Only the maximum poll interval can be controlled with settings, which is already much high (5 minutes) by default. This change gives users some configuration options so that they can adjust our batching behaviour to perform well on their workload, taking into account their specific throughput and latency requirements.

Closes #133

### CODE REVIEW

**Comment by jaley** [comment]
Note @lmars a couple of unrelated formatting changes have been picked up with the merge conflict resolution, shouldn't get in the way too much.

File: src/main/java/com/ably/kafka/connect/ChannelSinkTask.java [hunk 1]
```diff
@@ -77,25 +79,30 @@ public void start(Map<String, String> settings) {
         this.maxBufferLimit = Integer.parseInt(settings.getOrDefault(ChannelSinkConnectorConfig.BATCH_EXECUTION_MAX_BUFFER_SIZE,
                 ChannelSinkConnectorConfig.BATCH_EXECUTION_MAX_BUFFER_SIZE_DEFAULT));
 
+        this.maxBufferDelay = Long.parseLong(settings.getOrDefault(ChannelSinkConnectorConfig.BATCH_EXECUTION_MAX_BUFFER_DELAY_MS,
+            ChannelSinkConnectorConfig.BATCH_EXECUTION_MAX_BUFFER_DELAY_MS_DEFAULT));
+
+        this.buffer = new AutoFlushingBuffer<>(this.maxBufferDelay, this.maxBufferLimit, batch -> {
+            logger.info("SinkTask sending records: " + batch.size());
```
Comment (lmars) [inline src/main/java/com/ably/kafka/connect/ChannelSinkTask.java:86]: Should this be a debug log?

File: src/main/java/com/ably/kafka/connect/config/ChannelSinkConnectorConfig.java [hunk 2]
```diff
@@ -206,6 +206,10 @@ public class ChannelSinkConnectorConfig extends AbstractConfig {
 
     public static final String BATCH_EXECUTION_MAX_BUFFER_SIZE_DEFAULT = "1000";
 
+    public static final String BATCH_EXECUTION_MAX_BUFFER_DELAY_MS = "batchExecutionMaxBufferSizeMs";
```
Comment (lmars) [inline src/main/java/com/ably/kafka/connect/config/ChannelSinkConnectorConfig.java:209]: Similar to `BATCH_EXECUTION_MAX_BUFFER_SIZE_DOC` below, should there also be a `BATCH_EXECUTION_MAX_BUFFER_DELAY_MS_DOC` string documenting what this is?

Comment (jaley) [reply src/main/java/com/ably/kafka/connect/config/ChannelSinkConnectorConfig.java:212]: Yes, and actually it seems I also haven't added this new config to the config builder below. Added now.

**Review by lmars — APPROVED** [review]
LGTM

**Pull request merged**